saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/RelyingPartyRegistration.java

```diff
@@ -288,17 +393,80 @@ public Builder localEntityIdTemplate(String template) {
 			return this;
 		}
 
+		/**
+		 * Constructs a RelyingPartyRegistration object based on the builder configurations
+		 * @return a RelyingPartyRegistration instance
+		 */
 		public RelyingPartyRegistration build() {
 			return new RelyingPartyRegistration(
-					remoteIdpEntityId,
 					registrationId,
 					assertionConsumerServiceUrlTemplate,
-					idpWebSsoUrl,
+					new ProviderDetails(
+							providerDetailsBuilder.entityId,
+							providerDetailsBuilder.webSsoUrl,
+							providerDetailsBuilder.signAuthNRequest,
+							providerDetailsBuilder.binding
+					),
 					credentials,
 					localEntityIdTemplate
 			);
 		}
 	}
 
+	/**
+	 * Builder for IDP SSO endpoint configuration
+	 * @since 5.3
+	 */
+	public final static class ProviderDetailsBuilder {
```

Comment:
The builder should probably go inside `ProviderDetails`, simiar to how `RelyingPartyRegistration`'s builder is isn't of `RelyingPartyRegistration`. I think it should also be called `Builder` for consistency.

Reply:
Agreed.
c7937d150082ac8fd4a1ede9556876e10985d820